cetmix_tower_server/models/cx_tower_variable_value.py

```diff
@@ -87,6 +89,15 @@ class TowerVariableValue(models.Model):
         ),
     ]
 
+    @api.constrains("required", "value_char")
+    def _check_required_values(self):
+        for record in self.filtered(lambda r: r.required and r.active):
```

Comment:
"filtered" will perform and extra iteration over the same recorded.
So you can do the same check within the "for ..." cycle to save time and improve performance.

Reply:
You are correct that using filtered adds an extra iteration, which is suboptimal. The filtering logic can be incorporated directly into the for loop to avoid unnecessary operations. I will make the necessary changes to improve performance.

Reply:
I’ll keep in mind for the future that using filtered can be inefficient and it’s better to integrate filtering directly into the loop. However, this method is being removed as part of the current changes, so optimization won’t be necessary here.

Reply:
It's not inefficient, actually this is one of the most efficient methods of the ORM.
However in this case it will iterate the same recordset that you will iterate later. So there is simply no need for it because you can check the same conditions within your loop.